iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs

```diff
@@ -52,142 +52,113 @@ public static Exception GetExceptionFromOutcome(Outcome outcome)
             }
             else if (Equals(AmqpErrorCode.FramingError, amqpSymbol))
             {
-                return new IotHubClientException(message, true, amqpException);
+                retException = new(message, true, amqpException);
             }
             else if (Equals(AmqpErrorCode.ConnectionRedirect, amqpSymbol))
             {
-                return new IotHubClientException(message, true, amqpException);
+                retException = new(message, true, amqpException);
             }
             // AMQP Session Error
             else if (Equals(AmqpErrorCode.WindowViolation, amqpSymbol))
             {
-                return new IotHubClientException(message, true, amqpException);
+                retException = new(message, true, amqpException);
             }
             else if (Equals(AmqpErrorCode.ErrantLink, amqpSymbol))
             {
-                return new IotHubClientException(message, true, amqpException);
+                retException = new(message, true, amqpException);
             }
             else if (Equals(AmqpErrorCode.HandleInUse, amqpSymbol))
             {
-                return new IotHubClientException(message, true, amqpException);
+                retException = new(message, true, amqpException);
             }
             else if (Equals(AmqpErrorCode.UnattachedHandle, amqpSymbol))
             {
-                return new IotHubClientException(message, true, amqpException);
+                retException = new(message, true, amqpException);
             }
             // AMQP Link Error
             else if (Equals(AmqpErrorCode.DetachForced, amqpSymbol))
             {
-                return new IotHubClientException(message, true, amqpException);
+                retException = new(message, true, amqpException);
             }
             else if (Equals(AmqpErrorCode.TransferLimitExceeded, amqpSymbol))
             {
-                return new IotHubClientException(message, true, amqpException);
+                retException = new(message, true, amqpException);
             }
             else if (Equals(AmqpErrorCode.MessageSizeExceeded, amqpSymbol))
             {
-                return new IotHubClientException(message, IotHubClientErrorCode.MessageTooLarge, amqpException);
+                retException = new(message, IotHubClientErrorCode.MessageTooLarge, amqpException);
             }
             else if (Equals(AmqpErrorCode.LinkRedirect, amqpSymbol))
             {
-                return new IotHubClientException(message, true, amqpException);
+                retException = new(message, true, amqpException);
             }
             else if (Equals(AmqpErrorCode.Stolen, amqpSymbol))
             {
-                return new IotHubClientException(message, true, amqpException);
+                retException = new(message, true, amqpException);
             }
             // AMQP Transaction Error
```

Comment:
Does this comment need to move or be removed?

Reply:
It seems that this comment is for several transaction-error-related AmqpErrorCode in the following else-ifs, so no, we can just keep it as-is.